doc/manual.rst

```diff
@@ -4166,6 +4166,112 @@ For the above `openArray` and `var` rules, whether the expressions refers to
+
+Both `T` and `R` must be concrete types.
+
+If `R` is `void`, `blk` must be a statement. If `R` is not `void`, `blk` may
+either be an expression fitting type `R` or a statement.
+
+`blk` is typed within a new scope. The scope starts with containing a single
+symbol with type `(Ctx, P)`, with the name provided by the second argument,
+which must be an identifier.
+
+The `suspend` form must not statically appear:
+* inside a `try` block that has a `finally` clauses
+* in a statement list or expression after a `defer` statement
+* inside a `finally` or `except` clause
+* inside a `defer` block
+* inside a `static` or `const` context
+* as or within the third argument of a `suspend`
+* outside of a routine
+* as part of an `iterator` body
+* as part of a `.tailcall` routine body
+
+Saving the context refers to saving the value of locals (which includes owning
+parameters and the `result` variable, if any) in a context object. Views (both
+implicit and explicit) and non-owning parameters cannot be saved: if the
+continuation would needs them, a static error is reported. When a local that
+needs to be saved has a type that is non-copyable, a static error is reported.
+
+The value of the following locals need to be saved:
+* locals requiring cleanup and that store a value at the time the
+  `suspend` happens
+* locals whose value is read by the continuation
+
+Evaluation of `suspend`:idx: works as follows:
+1. owning locals whose value stored at the time of suspending is potentially
+   read by the suspend block (`blk`) have their value copied/moved as if
+   by assignment
+2. the local context needed to run the continuation is saved
+3. the saved context and the reified continuation are stored in `L`
+4. owning locals that are written by the suspend block without being read from
+   first are set to their default value (as if done by `wasMoved`)
+5. `blk` is evaluated as if it were the last statment or expression the
+   enclosing routine, ignoring exception handlers within the routine
+
+Taking the address of a local prior to suspending and then using the address
+for reading/writing in the suspend block (`blk`) or after resuming results
+in undefined behaviour.
+
+Resume
+~~~~~~
+
+Upon evaluating a call `x(y)` where `x` dynamically evaluates to `cont` and `y`
+dynamically evaluates to `ctx` (or a copy thereof), execution resumes in the
+suspended caller as if `suspend` returned.
+
+Upon evaluating a call `x(v, y)` where `x` dynamically evaluates to `cont` and
+`y` dynamically evaluates to `ctx` (or a copy thereof), execution resumes in
+the suspended caller as if `suspend` returned with value `v`.
+
+The value returned by a resumed callee is returned by the continuation
+invocation -- the same goes for raised exceptions.
+
+Copying
+~~~~~~~
+
+When copying a continuation instance, all saved values in the context object
+are copied, via a full copy, in the reverse order the locals they come from
+were declared.
```

Comment:
I'm not sure if defining a copy ordering is useful. This sounds like something implementations are free to decide on.

Reply:
For better or worse, copying usually implies destruction and is also allowed to have high-level effects (I/O, modification of globals, etc.). Given this circumstance, I'm of the opinion that the copy order should be well-defined.